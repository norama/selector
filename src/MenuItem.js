import React, { Component } from 'react';
import _ from 'underscore';

class MenuItem extends Component {

	render() {
		return (
			<div className='flex-container'>
				{this.props.option.icon && 
				 <div className='menuitem-icon'><img src={this.props.option.icon} alt='#' width='16' height='16'/></div>} 
				<div className='menuitem-label'>{this.props.option.render ? this.props.option.render() : this.props.option.label}</div>
			</div>);
	}
}

export default MenuItem;