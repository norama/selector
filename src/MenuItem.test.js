import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import MenuItem from './MenuItem';

it('renders with label only', () => {
	const option = {
		label: 'label',
		value: 'value'
	};
	const div = document.createElement('div');
	ReactDOM.render(<MenuItem option={option} />, div);
});
	
it('renders with label and icon', () => {
	const option = {
		label: 'label',
		value: 'value',
		icon: 'icon'
	};
	const div = document.createElement('div');
	ReactDOM.render(<MenuItem option={option} />, div);
});
	
it('component DOM matches snapshot', () => {
	const option = {
		label: 'label',
		value: 'value',
		icon: 'icon'
	};
	const component = renderer.create(
		<MenuItem option={option} />
	);
	let tree = component.toJSON();
	expect(tree).toMatchSnapshot();
});