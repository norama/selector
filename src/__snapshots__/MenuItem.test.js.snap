exports[`test component DOM matches snapshot 1`] = `
<div
  style={
    Object {
      "margin": "4px 4px 0px 4px",
    }
  }>
  <div
    style={
      Object {
        "display": "inline",
        "marginRight": "5px",
        "paddingTop": "5px",
      }
    }>
    <img
      alt="#"
      height="16"
      src="icon"
      width="16" />
  </div>
  <div
    style={
      Object {
        "display": "inline",
        "verticalAlign": "top",
      }
    }>
    label
  </div>
</div>
`;
